sql/core/src/main/scala/org/apache/spark/sql/jdbc/MsSqlServerDialect.scala

```diff
@@ -86,6 +86,35 @@ private object MsSqlServerDialect extends JdbcDialect {
       case "STDDEV_SAMP" => "STDEV"
       case _ => super.dialectFunctionName(funcName)
     }
+
+    override def build(expr: Expression): String = {
+      def isChildLikeExpression(child: Expression): Boolean = child match {
+        case gse: GeneralScalarExpression =>
+          gse.name() match {
+            case "STARTS_WITH" | "ENDS_WITH" | "CONTAINS" => true
+            case _ => false
+          }
+        case _ => false
+      }
+
+      // MsSqlServer does not support boolean comparison using standard comparison operators
+      // LIKE operator has a boolean result, hence we don't propagate these queries to MsSqlServer
+      expr match {
+        case e: GeneralScalarExpression =>
+          e.name() match {
+            case "=" | "<>" | "<=>" | "<" | "<=" | ">" | ">=" =>
```

Comment:
Please overwrite `visitBinaryComparison`

Reply:
Problem with overwriting `visitBinaryComparison` is that it takes strings as parameters. This leaves us with not much information about child expressions. Parsing these strings could be very complex and costly, so overriding this function is not the best option here.

Reply:
Shall we overwrite `visitBinaryComparison` and convert the boolean to integer type ? we can avoid the issue.